packages/maskbook/src/plugins/GoodGhosting/types.tsx

```diff
@@ -51,3 +52,8 @@ export interface LendingPoolData {
     poolAPY: BigNumber
     poolEarnings: BigNumber
 }
+
+export interface GameActionError {
```

Comment:
why not `extends Error`?

```ts
export class GameActionError extends Error {
    readonly state: TransactionStateType
    readonly transactionHash: string

    constructor(state: TransactionStateType, transactionHash: string) {
        super(state)
        this.state = state
        this.transactionHash = transactionHash
    }
}
```

Reply:
I don't need the Error class, as I don't use the name or message properties. I preferred a lightweight interface over a class. I could rename it to `GameActionState` instead of `GameActionError` if you think the naming is can be improved

Reply:
i want's you modify to class-based